### PR TITLE
pylint: Fix R1732 consider-using-with

### PR DESCRIPTION
**Pull request by stefanberger**
Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>

### CODE REVIEW

File: keylime/cmd_exec.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 
 def _execute(cmd, env=None, **kwargs):
-    proc = subprocess.Popen(cmd, env=env, stdout=subprocess.PIPE,
+    proc = subprocess.Popen(cmd, env=env, stdout=subprocess.PIPE,  #pylint: disable=consider-using-with
```
Comment (THS-on) [inline keylime/cmd_exec.py:15]: Why does with not work here?

File: keylime/keylime_agent.py [hunk 2]
```diff
@@ -279,9 +279,8 @@ def do_POST(self):
             shutil.rmtree("%s/unzipped" % secdir)
 
         # write out key file
-        f = open(secdir + "/" + self.server.enc_keyname, 'w', encoding="utf-8")
-        f.write(base64.b64encode(self.server.K).decode())
-        f.close()
+        with open(secdir + "/" + self.server.enc_keyname, 'w', encoding="utf-8") as f:
```
Comment (THS-on) [inline keylime/keylime_agent.py:282]: Can you fix the string formatting to `os.path.join`?.

I know that this is not directly related to this issue, but I try to convert them if I change the line anyway.

File: keylime/secure_mount.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 def check_mounted(secdir):
     """Inspect mountinfo to detect if a directory is mounted."""
     secdir_escaped = secdir.replace(" ", r"\040")
-    for line in open("/proc/self/mountinfo", "r", encoding="utf-8"):
+    for line in open("/proc/self/mountinfo", "r", encoding="utf-8"):  #pylint: disable=consider-using-with
```
Comment (THS-on) [inline keylime/secure_mount.py:23]: Can we not open the file with `with` and then iterate over `readline()`?

**Review by THS-on — COMMENTED** [review]
I do not see why the e2e are failing on this PR. Can you rebase on the latest master?

**Comment by THS-on** [comment]
The warning is called `R1732` not `W1732`.

File: keylime/cmd_exec.py (hunk 1 above)
Comment (stefanberger) [reply keylime/cmd_exec.py:15]: Due to the test failures I have been rolling things back that I am not sure about. I will actually start bisecting through the changes to see where the tests are failing. I don't see the reason. So I'll keep the CI/CD busy for a while...

File: keylime/keylime_agent.py (hunk 2 above)
Comment (stefanberger) [reply keylime/keylime_agent.py:282]: Fixed locally.

**Comment by stefanberger** [comment]
When testing https://github.com/keylime/keylime/pull/937/commits/447ef9f1de3cfa80028d1923e55e3b08a93534b1 there is one test-failure in testing-farm:fedora-35-x86_64:
```
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
::   -c delete
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

:: [ 10:34:35 ] :: [  BEGIN   ] :: Running 'lime_keylime_tenant -c delete'
Reading configuration from ['/etc/keylime.conf']
2022-03-23 10:34:36.296 - keylime.tpm - INFO - TPM2-TOOLS Version: 5.2
2022-03-23 10:34:36.302 - keylime.tenant - INFO - Setting up client TLS in /var/lib/keylime/cv_ca
2022-03-23 10:34:36.302 - keylime.tenant - WARNING - Using default UUID d432fbb3-d2f1-4a97-9ef7-75bd81c00000
2022-03-23 10:34:36.329 - keylime.tenant - INFO - Agent Info:
{"d432fbb3-d2f1-4a97-9ef7-75bd81c00000": {"operational_state": "Get Quote", "v": "5YiO9Kc1QK9h2OSdfiD0+uVbZ7imZhZmQsLcshl0QsY=", "ip": "127.0.0.1", "port": 9002, "tpm_policy": "{\"22\": [\"0000000000000000000000000000000000000001\", \"0000000000000000000000000000000000000000000000000000000000000001\", \"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001\", \"ffffffffffffffffffffffffffffffffffffffff\", \"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\", \"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\"], \"15\": [\"0000000000000000000000000000000000000000\", \"0000000000000000000000000000000000000000000000000000000000000000\", \"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\"], \"mask\": \"0x408400\"}", "vtpm_policy": "{\"23\": [\"ffffffffffffffffffffffffffffffffffffffff\", \"0000000000000000000000000000000000000000\"], \"15\": [\"0000000000000000000000000000000000000000\"], \"mask\": \"0x808000\"}", "meta_data": "{}", "allowlist_len": 6, "mb_refstate_len": 0, "accept_tpm_hash_algs": ["sha512", "sha384", "sha256", "sha1"], "accept_tpm_encryption_algs": ["ecc", "rsa"], "accept_tpm_signing_algs": ["ecschnorr", "rsassa"], "hash_alg": "sha1", "enc_alg": "rsa", "sign_alg": "rsassa", "verifier_id": "default", "verifier_ip": "127.0.0.1", "verifier_port": 8881, "severity_level": null, "last_event_id": null}}
2022-03-23 10:34:41.387 - keylime.tenant - ERROR - Timed out waiting for delete of agent d432fbb3-d2f1-4a97-9ef7-75bd81c00000 to complete at CV
:: [ 10:34:41 ] :: [   PASS   ] :: Command 'lime_keylime_tenant -c delete' (Expected 0, got 0)
:: [ 10:34:41 ] :: [   FAIL   ] :: File '/var/tmp/rlRun_LOG.AoxQaNdP' should contain 'INFO - CV completed deletion of agent d432fbb3-d2f1-4a97-9ef7-75bd81c00000' 
:: [ 10:34:41 ] :: [  BEGIN   ] :: Running 'lime_keylime_tenant -c cvstatus'
Reading configuration from ['/etc/keylime.conf']
2022-03-23 10:34:42.204 - keylime.tpm - INFO - TPM2-TOOLS Version: 5.2
2022-03-23 10:34:42.209 - keylime.tenant - INFO - Setting up client TLS in /var/lib/keylime/cv_ca
2022-03-23 10:34:42.209 - keylime.tenant - WARNING - Using default UUID d432fbb3-d2f1-4a97-9ef7-75bd81c00000
2022-03-23 10:34:42.227 - keylime.tenant - INFO - Agent Info:
{"d432fbb3-d2f1-4a97-9ef7-75bd81c00000": {"operational_state": "Get Quote", "v": "5YiO9Kc1QK9h2OSdfiD0+uVbZ7imZhZmQsLcshl0QsY=", "ip": "127.0.0.1", "port": 9002, "tpm_policy": "{\"22\": [\"0000000000000000000000000000000000000001\", \"0000000000000000000000000000000000000000000000000000000000000001\", \"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001\", \"ffffffffffffffffffffffffffffffffffffffff\", \"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\", \"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\"], \"15\": [\"0000000000000000000000000000000000000000\", \"0000000000000000000000000000000000000000000000000000000000000000\", \"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\"], \"mask\": \"0x408400\"}", "vtpm_policy": "{\"23\": [\"ffffffffffffffffffffffffffffffffffffffff\", \"0000000000000000000000000000000000000000\"], \"15\": [\"0000000000000000000000000000000000000000\"], \"mask\": \"0x808000\"}", "meta_data": "{}", "allowlist_len": 6, "mb_refstate_len": 0, "accept_tpm_hash_algs": ["sha512", "sha384", "sha256", "sha1"], "accept_tpm_encryption_algs": ["ecc", "rsa"], "accept_tpm_signing_algs": ["ecschnorr", "rsassa"], "hash_alg": "sha1", "enc_alg": "rsa", "sign_alg": "rsassa", "verifier_id": "default", "verifier_ip": "127.0.0.1", "verifier_port": 8881, "severity_level": null, "last_event_id": null}}
:: [ 10:34:42 ] :: [   PASS   ] :: Command 'lime_keylime_tenant -c cvstatus' (Expected 0, got 0)
:: [ 10:34:42 ] :: [   FAIL   ] :: File '/var/tmp/rlRun_LOG.gSGAYMcd' should contain 'Verifier at 127.0.0.1 with Port 8881 does not have agent d432fbb3-d2f1-4a97-9ef7-75bd81c00000' 
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
::   Duration: 7s
::   Assertions: 2 good, 2 bad
::   RESULT: FAIL (-c delete)
```

All other tests are passing. I prefer to ignore this test failure but something is flaky...

File: keylime/secure_mount.py (hunk 3 above)
Comment (stefanberger) [reply keylime/secure_mount.py]: This seems to work fine for me with python 3.10 just running this here:

```
>>> with open("/proc/self/mountinfo", "r", encoding="utf-8") as f:
...     for line in f:
...         print(line)
```

On the CI/CD system we now get this here:

```
Error in test.test_secure_mount.TestSecureMount.test_check_mounted_found
  File "/usr/lib64/python3.9/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib64/python3.9/unittest/case.py", line 592, in run
    self._callTestMethod(testMethod)
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
  File "/usr/lib64/python3.9/unittest/mock.py", line 1336, in patched
    return func(*newargs, **newkeywargs)
  File "/__w/keylime/keylime/test/test_secure_mount.py", line 31, in test_check_mounted_found
    self.assertTrue(secure_mount.check_mounted("/secdir"))
  File "/__w/keylime/keylime/keylime/secure_mount.py", line 23, in check_mounted
    with open("/proc/self/mountinfo", "r", encoding="utf-8") as f:
AttributeError: __enter__
```

Comment (THS-on) [reply keylime/secure_mount.py:23]: This is beacuse the open call is mocked, but the `__enter__` and `__exit__` functions are not: https://github.com/keylime/keylime/blob/master/test/test_secure_mount.py

Comment (stefanberger) [reply keylime/secure_mount.py:23]: So odd...
```
>>> subprocess.Popen.__enter__
<function Popen.__enter__ at 0x6fffffdf2d30>
>>> open.__enter__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'builtin_function_or_method' object has no attribute '__enter__'
>>> os.open.__enter__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'builtin_function_or_method' object has no attribute '__enter__'
```

Comment (THS-on) [reply keylime/secure_mount.py:23]: There seems to be a mocking object in the library for that: https://docs.python.org/3.7/library/unittest.mock.html#mock-open

Comment (stefanberger) [reply keylime/secure_mount.py]: Fixed in additional patch. What a struggle!

**Comment by stefanberger** [comment]
I wonder what that is when Fedora 34 fails and all other ones pass...

```
:: [ 19:42:43 ] :: [   FAIL   ] :: Command 'rlWaitForCmd 'tail /var/tmp/limeLib/agent.log | grep -q "Executing revocation action"' -m 10 -d 1 -t 10' (Expected 0, got 1)
```

**Comment by THS-on** [comment]
> I wonder what that is when Fedora 34 fails and all other ones pass...

@kkaarreell seems this like a timing issue?

@stefanberger just use `/packit test` to rerun the tests.

**Comment by kkaarreell** [comment]
> > I wonder what that is when Fedora 34 fails and all other ones pass...
> 
> @kkaarreell seems this like a timing issue?

I can increase the timeout... so far 10 seconds for revocation action initiation was enough. What should be the reasonable timeout?

**Comment by THS-on** [comment]
> I can increase the timeout... so far 10 seconds for revocation action initiation was enough. What should be the reasonable timeout?

10 seconds seem actually reasonable, but if it flaky again in the future we need to take a closer look what is causing this.

**Comment by kkaarreell** [comment]
Maybe some networking glitch on the actual virtual system... That would explain the above failure in the keylime tenant.
Having logs with timestamps would help. I will try to find a way how to incorporate more logs into the test output when test have failed.

**Comment by kkaarreell** [comment]
>  I will try to find a way how to incorporate more logs into the test output when test have failed.

Hopefully this would help next time
https://github.com/RedHat-SP-Security/keylime-tests/pull/70

**Review by mpeters — APPROVED** [review]
